common/achievement/get.ts

```diff
@@ -156,37 +156,32 @@ const assembleAchievementData = async (userAchievements: achievements_with_templ
-        if (evaluationResult) {
-            currentValue = dataAggregationKeys.map((key) => evaluationResult.resultObject[key]).reduce((a, b) => a + b, 0);
-            maxValue =
-                userAchievements[currentAchievementIndex].template.type === AchievementType.STREAK
-                    ? userAchievements[currentAchievementIndex].recordValue !== null && userAchievements[currentAchievementIndex].recordValue > currentValue
-                        ? userAchievements[currentAchievementIndex].recordValue
-                        : currentValue
-                    : dataAggregationKeys
-                          .map((key) => {
-                              // TODO: check if we can remove valueToAchieve
-                              return Number((userAchievements[currentAchievementIndex].template.conditionDataAggregations as any)[key].valueToAchieve || 0);
-                          })
-                          .reduce((a, b) => a + b, 0);
-            if (
-                userAchievements[currentAchievementIndex].template.type === AchievementType.STREAK &&
-                currentValue < maxValue &&
-                userAchievements[currentAchievementIndex].achievedAt
-            ) {
-                await prisma.user_achievement.update({
-                    where: { id: userAchievements[currentAchievementIndex].id },
-                    data: { achievedAt: null, isSeen: false },
-                });
-            }
+        currentValue = evaluationResult.aggregationResult;
+        maxValue = userAchievements[currentAchievementIndex].recordValue;
+
+        // Streaks are usually invalidated if a user does not show up for a certain meeting / course / etc.
+        // This means we don't have an event to reevaluate the streak naturally.
+        // There are two ways to solve this:
+        // 1. We could reevaluate the streak every time the user opens the app.
+        // 2. We could create a cronjob that reevaluates all streaks every now and then.
+        // As the cron would be more complex and less efficient, we chose the first option, which is implemented here:
+        if (evaluationResult.shouldInvalidateStreak) {
+            await prisma.user_achievement.update({
+                where: { id: userAchievements[currentAchievementIndex].id },
+                data: { achievedAt: null, isSeen: false },
+            });
         }
     }
+    if (userAchievements[currentAchievementIndex].template.type === AchievementType.TIERED) {
+        const evaluationResult = await isAchievementConditionMet(userAchievements[currentAchievementIndex]);
+        const {
+            template: { conditionDataAggregations },
+        } = userAchievements[currentAchievementIndex];
+        currentValue = evaluationResult.aggregationResult;
+        maxValue = Object.keys(conditionDataAggregations).reduce((acc, key) => acc + conditionDataAggregations[key].valueToAchieve, 0);
```

Comment:
Previously maxValue was set differently for STREAK / non-STREAK, no it is set for TIERED / non-TIERED? Can you add a comment here why we need to treat TIERED differently?

Reply:
That's not true, it's actually the same, but I shuffled the code a little bit.

Before:
```
if (isStreak or isTiered) {
  currentValue = evaluationResult;
  maxValue = isStreak ? recordValue : aggregate valueToAchieve;
}
```

Now:
```
if (isStreak) {
  currentValue = evaluationResult;
  maxValue =recordValue
}
if (isTiered) {
  currentValue = evaluationResult;
  maxValue = aggregate valueToAchieve;
}
```